src/persistence/common.jl

```diff
@@ -1,108 +1,82 @@
-NI(x...) = error("This function is not implemented.")
-
-# filemap is filled in the format specific source files
-const filemap = Dict{Symbol, Tuple{Function, Function, Function, Function}}()
-        # :gml        => (loadgml, loadgml_mult, savegml, savegml_mult)
-        # :graphml    => (loadgraphml, loadgraphml_mult, savegraphml, savegraphml_mult)
-        # :md         => (loadmatrixdepot, NOTIMPLEMENTED, NOTIMPLEMENTED, NOTIMPLEMENTED)
-
-# load a single graph by name
+abstract type AbstractGraphFormat end
 
 """
-    loadgraph(file, t=:lg)
+    loadgraph(file, gname="graph", format=LGFormat)
+
+Read a graph named `gname` from `file` in the format `format`.
 
-Read a graph from  `file` in the format `t`.
-Supported formats are `:lg`, `:gml`, `:dot`, `:graphml`, `:gexf`, `:net`, `:jld`, and `:graph6`.
+### Implementation Notes
+`gname` is graph-format dependent and is only used if the file contains
+multiple graphs; if the file format does not support multiple graphs, this
+value is ignored.
 """
-function loadgraph(fn::String, x...)
+function loadgraph(fn::AbstractString, gname::AbstractString, format::AbstractGraphFormat)
     GZip.open(fn,"r") do io
-        loadgraph(io, x...)
+        loadgraph(io, gname, format)
     end
 end
-
-function loadgraph(io::IO, t::Symbol=:lg)
-    t in keys(filemap) || error("Please select a supported graph format: one of $(keys(filemap))")
-    d = filemap[t][2](io)
-    return first(d)[2]
-end
+loadgraph(fn::AbstractString, gname::AbstractString="graph") = loadgraph(fn, gname, LGFormat())
 
 
 """
-    load(file, name, t=:lg)
+    loadgraphs(file, format=LGFormat)
 
-Load a graph with name `name` from `file` in format `t`.
+Load multiple graphs from `file` in the format `format`.
+Return a dictionary mapping graph name to graph.
 
-Currently supported formats are `:lg`, `:gml`, `:graphml`, `:gexf`, `:dot`, `:net`, and `graph6`.
+### Implementation Notes
+For unnamed graphs the default name \"graph\" will be used.
```

Comment:
Yeah, the whole docstring is wrong now since I changed the way we pass in file formats to the loaders / savers.